arm-recoveryservices/2016-06-01/swagger/registeredidentities.json

```diff
@@ -0,0 +1,536 @@
+            "AAD",
+            "AccessControlService",
+            "AzureActiveDirectory"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "AuthType",
+            "modelAsString": true
+          }
+        },
+        "certificate": {
+          "format": "byte",
+          "description": "The base64 encoded certificate raw data string",
+          "type": "string"
+        }
+      }
+    },
+    "ResourceCertificateAndAADDetails": {
+      "description": "Certificate details representing the Vault credentials for AAD.",
+      "required": [
+        "certificate",
+        "resourceId",
+        "aadAuthority",
+        "aadTenantId",
+        "servicePrincipalClientId",
+        "servicePrincipalObjectId",
+        "azureManagementEndpointAudience",
+        "subject",
+        "validFrom",
+        "validTo",
+        "thumbprint",
+        "friendlyName",
+        "issuer"
+      ],
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ResourceCertificateDetails"
+        }
+      ],
+      "properties": {
+        "certificate": {
+          "format": "byte",
+          "description": "The base64 encoded certificate raw data string.",
+          "type": "string"
+        },
+        "resourceId": {
+          "format": "int64",
+          "description": "Resource ID of the vault.",
+          "type": "integer"
+        },
+        "aadAuthority": {
+          "description": "AAD tenant authority.",
+          "type": "string"
+        },
+        "aadTenantId": {
+          "description": "AAD tenant Id.",
+          "type": "string"
+        },
+        "servicePrincipalClientId": {
+          "description": "AAD service principal clientId.",
+          "type": "string"
+        },
+        "servicePrincipalObjectId": {
+          "description": "AAD service principal ObjectId.",
+          "type": "string"
+        },
+        "azureManagementEndpointAudience": {
+          "description": "Azure Management Endpoint Audience.",
+          "type": "string"
+        },
+        "subject": {
+          "description": "Certificate Subject Name.",
+          "type": "string"
+        },
+        "validFrom": {
+          "format": "date-time",
+          "description": "Certificate Validity start Date time.",
+          "type": "string"
+        },
+        "validTo": {
+          "format": "date-time",
+          "description": "Certificate Validity End Date time.",
+          "type": "string"
+        },
+        "thumbprint": {
+          "description": "Certificate thumbprint.",
+          "type": "string"
+        },
+        "friendlyName": {
+          "description": "Certificate friendlyname.",
+          "type": "string"
+        },
+        "issuer": {
+          "description": "Certificate issuer.",
+          "type": "string"
+        }
+      },
+      "x-ms-discriminator-value": "AzureActiveDirectory"
+    },
+    "ResourceCertificateAndACSDetails": {
```

Comment:
Is this being used/allOfed anywhere? What is the discriminator field here given you are setting the x-ms-discriminator extension